src/commands/cmd_tdigest.cc

```diff
@@ -0,0 +1,146 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "command_parser.h"
+#include "commander.h"
+#include "server/redis_reply.h"
+#include "server/server.h"
+#include "status.h"
+#include "string_util.h"
+#include "types/redis_tdigest.h"
+
+namespace redis {
+namespace {
+constexpr auto kCompressionArg = "compression";
+constexpr auto kErrWrongKeyword = "wrong keyword";
+constexpr auto kErrWrongNumOfArguments = "wrong number of arguments";
+constexpr auto kErrParseCompression = "error parsing compression parameter";
+constexpr auto kErrCompressionMustBePositive = "compression parameter needs to be a positive integer";
+const static auto kErrCompressionOutOfRange =
+    "compression must be between 1 and " + std::to_string(kTDigestMaxCompression);
+constexpr auto kErrKeyNotFound = "key does not exist";
+constexpr auto kErrKeyAlreadyExists = "key already exists";
+
+constexpr auto kInfoCompression = "Compression";
+constexpr auto kInfoCapacity = "Capacity";
+constexpr auto kInfoMergedNodes = "Merged nodes";
+constexpr auto kInfoUnmergedNodes = "Unmerged nodes";
+constexpr auto kInfoMergedWeight = "Merged weight";
+constexpr auto kInfoUnmergedWeight = "Unmerged weight";
+constexpr auto kInfoObservations = "Observations";
+constexpr auto kInfoTotalCompressions = "Total compressions";
+}  // namespace
+
+class CommandTDigestCreate : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    CommandParser parser(args, 1);
+    key_name_ = GET_OR_RET(parser.TakeStr());
+    options_.compression = 100;
+    if (parser.EatEqICase(kCompressionArg)) {
+      if (!parser.Good()) {
+        return {Status::RedisParseErr, kErrWrongNumOfArguments};
+      }
+      auto status = parser.TakeInt<int32_t>();
+      if (!status) {
+        return {Status::RedisParseErr, kErrParseCompression};
+      }
+      auto compression = *status;
+      if (compression <= 0) {
+        return {Status::RedisParseErr, kErrCompressionMustBePositive};
+      }
+      if (compression < 1 || compression > static_cast<int32_t>(kTDigestMaxCompression)) {
+        return {Status::RedisParseErr, kErrCompressionOutOfRange};
+      }
+      options_.compression = static_cast<uint32_t>(compression);
+    }
+    if (parser.Good()) {
+      parser.RawTake();
+      return {Status::RedisParseErr, parser.Good() ? kErrWrongKeyword : kErrWrongNumOfArguments};
```

Comment:
```suggestion
      return {Status::RedisParseErr, kErrWrongNumOfArguments};
```

Reply:
Hi @PragmaTwice ,

Thanks for your suggestion! 😊

I have simplified the logic as suggested.

Best Regards,
Edward